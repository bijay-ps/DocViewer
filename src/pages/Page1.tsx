import { ZuiButton } from '@zeiss/zui-react';
import type { FC } from 'react';

import GlobalStyle from '../styles';

export const Page1: FC<{ cb?: () => void }> = ({ cb }) => {
  function clickMe(): void {
    if (cb) {
      cb();
    }
  }

  return (
    <>
      <GlobalStyle />
      {/*<iframe src="https://mozilla.github.io/pdf.js/web/viewer.html?file=https://api.slingacademy.com/v1/sample-data/files/just-text.pdf" width="1920" height="830" />*/}
      <iframe src="https://mozilla.github.io/pdf.js/web/viewer.html?file=https://mystrgacc23.blob.core.windows.net/mycontainer/Wick_John_19931001_20210901_1229_PDF_OS_urn_oid_1.2.276.0.75.2.5.80.25.3.210901123311149.123926587919512.4158555.pdf?sp=r&st=2024-05-30T04:48:25Z&se=2025-05-30T12:48:25Z&spr=https&sv=2022-11-02&sr=b&sig=teGcLnrIcgTjJlm3gaAAAbA8MVmIj8BJVmI0h1wPp1A%3D" width="1920" height="830" />
      {/*<iframe src="http://localhost:8888/web/viewer.html?file=compressed.tracemonkey-pldi-09.pdf" width="1920" height="830" />*/}
    </>
  );
};
