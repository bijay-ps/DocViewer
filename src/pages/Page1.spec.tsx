import { fireEvent, render, screen } from '@testing-library/react';

import { Page1 } from './Page1';

describe('Page1', () => {
  it('should render the greeting', () => {
    render(<Page1 />);

    expect(screen.getByText('Hello React Page 1!')).toBeInTheDocument();
  });

  it('should render the element', () => {
    const { container } = render(<Page1 />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should fire callback on button click', () => {
    const callback = jest.fn();
    render(<Page1 cb={callback} />);

    fireEvent.click(screen.getByText('Click me!'));

    expect(callback).toHaveBeenCalledTimes(1);
  });
});
