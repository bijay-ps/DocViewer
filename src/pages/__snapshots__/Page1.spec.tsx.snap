// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page1 should render the element 1`] = `
.c0 {
  padding: 10px;
  border: 1px solid #999;
  margin: 10px;
}

<div
  class="c0"
>
  <p>
    Hello React Page 1!
  </p>
  <zui-button>
    Click me!
  </zui-button>
  <p
    class="my-pilet-error"
  >
    This in an error styled by Pilet global CSS in src/index.css.
  </p>
</div>
`;
